flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/DefaultExecutionGraphFactory.java

```diff
@@ -159,6 +160,11 @@ public ExecutionGraph createAndRestoreExecutionGraph(
                     }
                 };
 
+        int totalNumberOfSubTasks =
+                StreamSupport.stream(jobGraph.getVertices().spliterator(), false)
+                        .mapToInt(jobVertex -> jobVertex.getParallelism())
```

Comment:
We need to read the DoP from the store to account for rescaling; 
otherwise, on downscaling, we'll get the old (high) DoP and won't report downscaling: 
```suggestion
                        .mapToInt(
                                jobVertex ->
                                        vertexParallelismStore
                                                .getParallelismInfo(jobVertex.getID())
                                                .getParallelism())
```